nanoprintf.h

```diff
@@ -744,7 +744,7 @@ int npf_vpprintf(npf_putc pc, void *pc_ctx, char const *format, va_list args) {
         cbuf = va_arg(args, char *);
 #if NANOPRINTF_USE_PRECISION_FORMAT_SPECIFIERS == 1
         for (char const *s = cbuf;
-             *s && ((fs.prec_opt == NPF_FMT_SPEC_OPT_NONE) || (cbuf_len < fs.prec));
+             s && *s && ((fs.prec_opt == NPF_FMT_SPEC_OPT_NONE) || (cbuf_len < fs.prec));
```

Comment:
I think I'd prefer a rearrangement to a new clause:
```suggestion
             ((fs.prec_opt == NPF_FMT_SPEC_OPT_NONE) || (cbuf_len < fs.prec)) && *s;
```

That way the code footprint should stay the same, and this specific change stays minimal. I'd prefer to do "nanoprintf gracefully handles null strings" in a different PR after giving it some thought- we might want to print `<null>` or something similar.

Reply:
yeah thats more elegant, I've made the change.